src/lib/markets/graph.ts

```diff
@@ -709,6 +669,162 @@ export async function fetchSubPositions({
   return { changes, hasMore: updateds.length === first }
 }
 
+export async function fetchTradeHistory({
+  graphClient,
+  address,
+  fromTs,
+  toTs,
+}: {
+  graphClient: GraphQLClient
+  address: Address
+  fromTs?: bigint
+  toTs?: bigint
+}) {
+  const maxTimeRange = Day * 7n
+  const now = BigInt(nowSeconds())
+  if (!fromTs) fromTs = now - maxTimeRange
+  if (!toTs) toTs = now
+
+  if (toTs - fromTs > maxTimeRange) {
+    throw new Error('The time range exceeds the maximum allowed range of 7 days.')
+  }
+  // TODO(arjun): update this query to use trades entity in subgraph when available
+  const tradeHistoryQuery = gql(`
+  query fetchTradeHistory($account: Bytes!, $fromTs: BigInt, $toTs: BigInt, $first: Int!, $skip: Int!) {
+   accountPositionProcesseds(
+     where: { account: $account, blockTimestamp_gte: $fromTs, blockTimestamp_lte: $toTs },
```

Comment:
We use `toOracleVersion` instead of `blockTimestamp` here (it should match whatever we're ordering by)